x/pool-incentives/keeper/keeper.go

```diff
@@ -162,6 +161,29 @@ func (k Keeper) GetPoolIdFromGaugeId(ctx sdk.Context, gaugeId uint64, lockableDu
 	return sdk.BigEndianToUint64(bz), nil
 }
 
+// GetGaugesForCFMMPool returns the gauges associated with the given CFMM pool ID, by first retrieving
+// the lockable durations for the pool, then using them to query the pool incentives keeper for the
+// gauge IDs associated with each duration, and finally using the incentives keeper to retrieve the
+// actual gauges from the retrieved gauge IDs.
+func (k Keeper) GetGaugesForCFMMPool(ctx sdk.Context, poolId uint64) ([]incentivestypes.Gauge, error) {
```

Comment:
it only works for a CFMM pool because CL pools have only 1 gauge.

Added a contract here:
```
// CONTRACT: pool id must be assocated with a CFMM pool.
```